Admin/Test-AMSI.ps1

```diff
@@ -1,144 +1,793 @@
+                                            ($testTime.ToUniversalTime().Subtract($DateTime) -gt - $marginTime ) ) {
+                                            if ( ($line.UrlHost.ToLower() -eq $server.ToLower() ) -and
+                                                ($line.UrlStem.ToLower() -eq '/ecp/x.js'.ToLower() ) -and
+                                                ($line.ScanResult.ToLower() -eq 'Detected'.ToLower() ) ) {
+                                                Write-Host ""
+                                                Write-Host "We found a detection in HttpRequestFiltering logs: " -ForegroundColor Green
+                                                Write-Host "$line"
+                                                $detected = $true
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        } while ( ( -not $detected ) -and ((Get-Date) -lt $timeout) )
+                        if ((Get-Date) -ge $timeout) {
+                            Write-Warning  "We have not found activity on the server in the last minute."
+                        }
+                        if ( -not $detected ) {
+                            Write-Warning "We have not found a detection."
+                        }
+                    } else {
+                        Write-Host "We could not access HttpRequestFiltering folder on $server" -ForegroundColor Red
+                    }
+                } else {
+                    Write-Host "Check your log files located in %ExchangeInstallPath%\Logging\HttpRequestFiltering\ in all server that provide $server endpoint"
+                }
+            } elseif ( $_.Exception.Message -eq "The remote server returned an error: (500) Internal Server Error.") {
+                $host.ui.RawUI.ForegroundColor = "Red"
+                Write-Host $msgNewLine
+                Write-Host $Message
+                Write-Host $msgNewLine
+                $host.ui.RawUI.ForegroundColor = "Yellow"
+                Write-Host "If you are using Microsoft Defender, RealTime protection could be disabled or then AMSI may be disabled."
+                Write-Host "If you are using a 3rd Party AntiVirus Product that may not be AMSI capable (Please Check with your AntiVirus Provider for Exchange AMSI Support)"
+                $host.ui.RawUI.ForegroundColor = $currentForegroundColor
+            } elseif ( $_.Exception.Message.StartsWith("The remote name could not be resolved:") ) {
+                $host.ui.RawUI.ForegroundColor = "Red"
+                Write-Host $msgNewLine
+                Write-Host $Message
+                Write-Host $msgNewLine
+                $host.ui.RawUI.ForegroundColor = $currentForegroundColor
+            } else {
+                $host.ui.RawUI.ForegroundColor = "Red"
```

Comment:
We should get rid of the `$host.ui.RawUI.ForegroundColor` stuff in the script and instead, simply do a `Write-Host -ForegroundColor Red`

Reply:
It is something that comes from the original version but I think that David suggested to do it in another PR.

Reply:
Yep, we can do this in a different PR before we make another change to the script that isn't a major issue.